### PR TITLE
Resolved multiple issues related to the Test Runs.

### PR DESCRIPTION
**Pull request by deyordanov**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1422
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1287
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1377
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1555

### CODE REVIEW

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Data/OJS.Data.Models/Tests/TestRun.cs [hunk 1]
```diff
@@ -1,33 +1,31 @@
...
+    public virtual Test Test { get; set; } = null!;
 
-        public int TimeUsed { get; set; }
+    public int TimeUsed { get; set; }
 
-        public long MemoryUsed { get; set; }
+    public long MemoryUsed { get; set; }
 
-        public TestRunResultType ResultType { get; set; }
+    public int ResultType { get; set; }
```
[the remark this answers is not in the document]
Comment (deyordanov) [reply Data/OJS.Data.Models/Tests/TestRun.cs:20]: For consistency, I could revert it back to the enum type.

[the remark this answers is not in the document]
Comment (gogo4ds) [reply Data/OJS.Data.Models/Tests/TestRun.cs:20]: Enum is better and conversion to int should be trivial.

[the remark this answers is not in the document]
Comment (deyordanov) [reply Data/OJS.Data.Models/Tests/TestRun.cs]: Done.

File: Services/Administration/OJS.Services.Administration.Business/Submissions/SubmissionsBusinessService.cs [hunk 2]
```diff
@@ -90,7 +90,7 @@ public Task RecalculatePointsByProblem(int problemId)
                         s.Id,
                         s.ParticipantId,
                         CorrectTestRuns = s.TestRuns.Count(t =>
-                            t.ResultType == TestRunResultType.CorrectAnswer &&
+                            t.ResultType == (int)TestRunResultType.CorrectAnswer &&
```
Comment (gogo4ds) [inline Services/Administration/OJS.Services.Administration.Business/Submissions/SubmissionsBusinessService.cs:93]: Do you need the cast?

File: Services/Worker/OJS.Services.Worker.Business/Extensions/TaskResultServiceModelExtensions.cs [hunk 3]
```diff
@@ -25,7 +25,7 @@ public static void CalculatePoints(this TaskResultServiceModel? taskResult, int
         {
             var correctAnswersCount = testResults
                 .Count(x =>
-                    Enum.Parse<TestRunResultType>(x.ResultType) == TestRunResultType.CorrectAnswer);
+                    x.ResultType == (int)TestRunResultType.CorrectAnswer);
```
Comment (gogo4ds) [inline Services/Worker/OJS.Services.Worker.Business/Extensions/TaskResultServiceModelExtensions.cs:28]: And here?

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Services/Administration/OJS.Services.Administration.Business/Submissions/SubmissionsBusinessService.cs (hunk 2 above)
Comment (deyordanov) [reply Services/Administration/OJS.Services.Administration.Business/Submissions/SubmissionsBusinessService.cs]: Done.

File: Services/Worker/OJS.Services.Worker.Business/Extensions/TaskResultServiceModelExtensions.cs (hunk 3 above)
Comment (deyordanov) [reply Services/Worker/OJS.Services.Worker.Business/Extensions/TaskResultServiceModelExtensions.cs]: Done.

**Review by gogo4ds — DISMISSED** [review]

**Review by gogo4ds — APPROVED** [review]

**Pull request merged**